### PR TITLE
Spec for opcode `DELEGATECALL`

### PR DESCRIPTION
**Pull request by silathdiir**
Related upstream issue - https://github.com/privacy-scaling-explorations/zkevm-specs/issues/293

Referenced [go-ethereum DELEGATECALL](https://github.com/ethereum/go-ethereum/blob/2b1299b1c006077c56ecbad32e79fc16febe3dd6/core/vm/evm.go#L323) and [function asDelegate](https://github.com/ethereum/go-ethereum/blob/2b1299b1c006077c56ecbad32e79fc16febe3dd6/core/vm/contract.go#L137).

### CODE REVIEW

File: src/zkevm_specs/evm/execution/callop.py [hunk 1]
```diff
@@ -16,18 +16,22 @@
...
+        opcode, (Opcode.CALL, Opcode.DELEGATECALL, Opcode.STATICCALL)
+    )
     instruction.responsible_opcode_lookup(opcode)
 
     callee_call_id = instruction.curr.rw_counter
 
     tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
     reversion_info = instruction.reversion_info()
     caller_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+    parent_caller_address = instruction.call_context_lookup(CallContextFieldTag.CallerAddress)
```
Comment (DreamWuGit) [inline src/zkevm_specs/evm/execution/callop.py:31]: seems make sense to add condition when is_delegatecall ==1 , then do these two  new lookups.

File: src/zkevm_specs/evm/execution/callop.py [hunk 2]
```diff
@@ -16,18 +16,22 @@
...
+    parent_value = instruction.call_context_lookup(CallContextFieldTag.Value)
```
Comment (DreamWuGit) [inline src/zkevm_specs/evm/execution/callop.py:32]: `parent_call_value` name better ?

**Comment by DreamWuGit** [comment]
others LGTM!

File: src/zkevm_specs/evm/execution/callop.py (hunk 1 above)
Comment (silathdiir) [reply src/zkevm_specs/evm/execution/callop.py]: Fixed to check `is_delegatecall == 1` for both `parent_caller_address` and `parent_call_value`:
```
parent_caller_address = instruction.select(
    is_delegatecall, instruction.call_context_lookup(CallContextFieldTag.CallerAddress), FQ(0)
)
parent_call_value = instruction.select(
    is_delegatecall, instruction.call_context_lookup(CallContextFieldTag.Value), FQ(0)
)
```

File: src/zkevm_specs/evm/execution/callop.py (hunk 2 above)
Comment (silathdiir) [reply src/zkevm_specs/evm/execution/callop.py]: Renamed `parent_value` to `parent_call_value`.

**Comment by silathdiir** [comment]
Thanks @DreamWuGit help code review. Moved to upstream - https://github.com/privacy-scaling-explorations/zkevm-specs/pull/294

**Pull request closed without merging**